android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java

```diff
@@ -2437,6 +2437,21 @@ public void testInitializationInternalTable() throws InterruptedException, IOExc
         }
     }
 
+    @Test
+    public void testHashCode() throws InterruptedException, IOException {
+        DataStore<Person> store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        client.getSyncManager().clear(Person.COLLECTION);
+        clearBackend(store);
+        Person person = createPerson(TEST_USERNAME);
+        DefaultKinveyClientCallback saveCallback = save(store, person);
+        assertNotNull(saveCallback.result);
+        assertNull(saveCallback.error);
+        store.syncBlocking(null);
+        List<Person> personList = store.find();
+        Person person1 = personList.get(0);
+        assertNotNull(person1.hashCode());
```

Comment:
I think a good addition to this test would be to perform multiple finds in a row on the same object, and verify that the hash code is consistently the same.

Reply:
Updated